spec/01-unit/016-dns_spec.lua

```diff
@@ -0,0 +1,123 @@
+-- simple debug function
+local dump = function(...)
+  print(require("pl.pretty").write({...}))
+end
+
+
+describe("DNS", function()
+  local balancer, resolver, query_func, old_new
+  local mock_records, singletons, client
+
+
+  setup(function()
+    stub(ngx, "log")
+    _G._TEST = true
```

Comment:
We don't seem to be using the features enabled by this flag (https://github.com/Kong/lua-resty-dns-client/blob/2f6b954a7b4b9a18629f973aa7d0d443b99aa7b0/src/resty/dns/client.lua#L1377-L1380)? If not, let's avoid this pattern. I tried this locally and the tests succeeded. Generally, I would consider this bad practices since this pattern can easily be abused and hide different behaviors in "test" vs "non-test" modes, resulting in faulty/false positive unit tests. I am sure you are already aware anyway

Reply:
good catch. Left over, initially the tests did use it.